api/processor.js

```diff
@@ -404,9 +407,11 @@ function nonEmptyStr (str) {
   }
 
 // allow A-z0-9._-, dont allow a trailing .
-var badNameCharsRegex = /[^A-z0-9\._-]/g
-function makeNameSafe (str) {
-  str = str.replace(badNameCharsRegex, '_')
+var badNameCharsRegex = /[^A-Za-z0-9\._-]/g
```

Comment:
Any good way of refactoring this regexp into a function? Seems it's used at five places.